tests/kernel/sched/schedule_api/src/user_api.c

```diff
@@ -78,4 +86,266 @@ void test_user_k_is_preempt(void)
+	k_thread_resume(NULL);
+
+	/* should not go here */
+	ztest_test_fail();
+}
+
+/**
+ * @brief Test k_thread_resume() API
+ *
+ * @details Create a thread and set k_thread_resume() para input to NULL
+ * will trigger fatal error.
+ *
+ * @ingroup kernel_sched_tests
+ *
+ * @see k_thread_resume()
+ */
+void test_k_thread_resume_init_null(void)
+{
+	k_tid_t tid = k_thread_create(&user_thread, ustack, STACK_SIZE,
+			(k_thread_entry_t)thread_resume_init_null,
+			NULL, NULL, NULL,
+			K_PRIO_PREEMPT(THREAD_TEST_PRIORITY),
+			K_USER | K_INHERIT_PERMS, K_NO_WAIT);
+
+	k_thread_join(tid, K_FOREVER);
+}
+#else
+void test_k_thread_resume_init_null(void)
+{
+	ztest_test_skip();
+}
+#endif
+
+#ifdef CONFIG_USERSPACE
+static void thread_priority_get_init_null(void *p1, void *p2, void *p3)
+{
+	ztest_set_fault_valid(true);
+	k_thread_priority_get(NULL);
+
+	/* should not go here */
+	ztest_test_fail();
+}
+
+/**
+ * @brief Test k_thread_priority_get() API
+ *
+ * @details Create a thread and set thread_k_thread_priority_get() para input to NULL
+ *
+ * @ingroup kernel_sched_tests
+ *
+ * @see thread_k_thread_priority_get()
+ */
+void test_k_thread_priority_get_init_null(void)
+{
+	k_tid_t tid = k_thread_create(&user_thread, ustack, STACK_SIZE,
+			(k_thread_entry_t)thread_priority_get_init_null,
+			NULL, NULL, NULL,
+			K_PRIO_PREEMPT(THREAD_TEST_PRIORITY),
+			K_USER | K_INHERIT_PERMS, K_NO_WAIT);
+
+	k_thread_join(tid, K_FOREVER);
+}
+#else
+void test_k_thread_priority_get_init_null(void)
+{
+	ztest_test_skip();
+}
+#endif
+
+#ifdef CONFIG_USERSPACE
+static void thread_priority_set_init_null(void *p1, void *p2, void *p3)
+{
+	ztest_set_fault_valid(true);
+	k_thread_priority_set(NULL, 0);
+
+	/* should not go here */
+	ztest_test_fail();
+}
+
+/**
+ * @brief Test k_thread_priority_set() API
+ *
+ * @details Create a thread and set k_thread_priority_set() para input to NULL
+ *
+ * @ingroup kernel_sched_tests
+ *
+ * @see k_thread_priority_set()
+ */
+void test_k_thread_priority_set_init_null(void)
+{
+	k_tid_t tid = k_thread_create(&user_thread, ustack, STACK_SIZE,
+			(k_thread_entry_t)thread_priority_set_init_null,
+			NULL, NULL, NULL,
+			K_PRIO_PREEMPT(THREAD_TEST_PRIORITY),
+			K_USER | K_INHERIT_PERMS, K_NO_WAIT);
+
+	k_thread_join(tid, K_FOREVER);
+}
+#else
+void test_k_thread_priority_set_init_null(void)
+{
+	ztest_test_skip();
+}
+#endif
+
+#ifdef CONFIG_USERSPACE
+static void thread_priority_set_invalid1(void *p1, void *p2, void *p3)
+{
+	ztest_set_fault_valid(true);
+
+	/* set valid priority value outside the priority range will invoke fatal error */
+	k_thread_priority_set(k_current_get(), K_LOWEST_APPLICATION_THREAD_PRIO + 1);
+
+	/* should not go here */
+	ztest_test_fail();
+}
+
+/**
+ * @brief Test k_thread_priority_set() API
+ *
+ * @details Check input para range fail in userspace test.
+ *
+ * @ingroup kernel_sched_tests
+ *
+ * @see k_thread_priority_set()
+ */
+void test_k_thread_priority_set_invalid1(void)
```

Comment:
Maybe we can name it better, such as "test_k_thread_priority_set_overmax" or something more meaningful than invalid1.   So are the following test cases.

Reply:
Hi, enjia.  I've changed the function name.